dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/schema/JsonbMappingTest.java

```diff
@@ -509,4 +512,200 @@ public void testDefaultMethod() {
+
+        public ContainerMap() {}
+
+        /**
+         * @param bm
+         */
+        public ContainerMap(BoxMap<X, String, Integer> bm) {
+            this.bm = bm;
+        }
+
+        /**
+         * @return the bm
+         */
+        public BoxMap<X, String, Integer> getBm() {
+            return bm;
+        }
+
+        /**
+         * @param bm the bm to set
+         */
+        public void setBm(BoxMap<X, String, Integer> bm) {
+            this.bm = bm;
+        }
+
+    }
+
+    public static class ContainerConcrete {
+        public ContainerMap<String> cm;
+
+        /**
+         */
+        public ContainerConcrete() {}
+
+        /**
+         * @param cm
+         */
+        public ContainerConcrete(ContainerMap<String> cm) {
+            this.cm = cm;
+        }
+
+        /**
+         * @return the cm
+         */
+        public ContainerMap<String> getCm() {
+            return cm;
+        }
+
+        /**
+         * @param cm the cm to set
+         */
+        public void setCm(ContainerMap<String> cm) {
+            this.cm = cm;
+        }
+
+    }
+
+    @Test
+    public void testGeneric() {
+        BoxMap<String, String, Integer> bm = new BoxMap<>("1str", "2str", 3);
+        JSONAssert.assertEquals("{\"var1\":\"1str\",\"var2\":\"2str\",\"var3\":3}", jsonb.toJson(bm), true);
+        BoxMap<String, String, Integer> bmIn = jsonb.fromJson("{\"var1\":\"1str\",\"var2\":\"2str\",\"var3\":3}", new TypeLiteral<BoxMap<String, String, Integer>>() {
+            private static final long serialVersionUID = 1L;
+        }.getType());
+//        JSONAssert.assertEquals("{\"var1\":\"1str\",\"var2\":\"2str\",\"var3\":3}", jsonb.toJson(bmIn), true);
+//        JSONAssert.assertEquals("{\"type\":\"object\",\"properties\":{\"var3\":{\"type\":\"object\"},\"var2\":{\"type\":\"object\"},\"var1\":{\"type\":\"object\"}},\"required\":[\"var3\",\"var2\",\"var1\"]}",
+//                                registry.getSchema(BoxMap.class, SchemaDirection.INPUT).toString(), true);
+
+        ContainerMap<String> cm = new ContainerMap<>(bm);
+        JSONAssert.assertEquals("{\"bm\":{\"var1\":\"1str\",\"var2\":\"2str\",\"var3\":3}}", jsonb.toJson(cm), true);
+        ContainerMap<String> cmIn = jsonb.fromJson("{\"bm\":{\"var1\":\"1str\",\"var2\":\"2str\",\"var3\":3}}", new TypeLiteral<ContainerMap<String>>() {
+            private static final long serialVersionUID = 1L;
+        }.getType());
+//        JSONAssert.assertEquals("{\"bm\":{\"var1\":\"1str\",\"var2\":\"2str\",\"var3\":3}}", jsonb.toJson(cmIn), true);
+//        JSONAssert.assertEquals("{\"type\":\"object\",\"properties\":{\"bm\":{\"type\":\"object\",\"properties\":{\"var3\":{\"type\":\"integer\"},\"var2\":{\"type\":\"string\"},\"var1\":{\"type\":\"object\"}},\"required\":[\"var3\",\"var2\",\"var1\"]}},\"required\":[\"bm\"]}",
+//                                registry.getSchema(ContainerMap.class, SchemaDirection.INPUT).toString(), true);
```

Comment:
This commented out code should be removed (or fixed and uncommented).

Do as a follow-up.